### PR TITLE
[CFNetwork] Fix ownership rules of CFHTTPAuthentication

### PR DESCRIPTION
**Pull request by Therzok**
`Create` methods give ownership of the object, so in this case, owns=true. Prevents leak of CFHTTPAuthentication

### CODE REVIEW

**Comment by Therzok** [comment]
Are there any tests needed? CFRetainCount based?

**Comment by Therzok** [comment]
`MonoTouchFixtures.CoreServices.HttpMessageTest.CreateResponseAuth: System.NullReferenceException : Object reference not set to an instance of an object`

What would be a good way to get a HttpMessage response that's valid?

**Comment by mandel-macaque** [comment]
@Therzok you could do something similar to what we do in other network tests, create a real request and get the message from there, using `CreateRequest (CoreFoundation.CFUrl url, Foundation.NSString method, Version version);` For example: https://github.com/xamarin/xamarin-macios/blob/master/tests/monotouch-test/System.Net.Http/MessageHandlers.cs#L80

You can create a request against httpbin.org (we will at some point host this in a container for our tests). Httpbin.org has a nice API in case you want to test auth or other network related methods.

**Comment by Therzok** [comment]
If anyone can take this over, please do. Currently focusing on other tasks, will probably be able to get back to it in 2 weeks.

File: tests/monotouch-test/CoreServices/HttpMessageTest.cs [hunk 1]
```diff
@@ -62,14 +67,47 @@ public void CreateEmptyFalse ()
...
+			var taskCompletionSource = new TaskCompletionSource<CFHTTPMessage> ();
+			// the following code has to be in a diff thread, else, we are blocking the current loop, not cool
+			// perform a request so that we fail in the auth, then create the auth object and check the count
+			TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () => {
+				using (var request = CFHTTPMessage.CreateRequest (
+					new Uri (NetworkResources.Httpbin.GetStatusCodeUrl (HttpStatusCode.Unauthorized)), "GET", null)) {
+					request.SetBody (Array.Empty<byte> ()); // empty body, we are not interested
+					using (var stream = CFStream.CreateForHTTPRequest (request)) {
+						Assert.IsNotNull (stream, "Null stream");
+						// we are nonly interested in the completed event
```
Comment (rolfbjarne) [inline tests/monotouch-test/CoreServices/HttpMessageTest.cs]:
```suggestion
						// we are only interested in the completed event
```

File: tests/monotouch-test/CoreServices/HttpMessageTest.cs [hunk 2]
```diff
@@ -62,14 +67,47 @@ public void CreateEmptyFalse ()
...
-			using (var m = CFHTTPMessage.CreateRequest (new Uri ("http://www.xamarin.com"), "GET", new Version (1, 0))) {
+			using (var m = CFHTTPMessage.CreateRequest (NetworkResources.XamarinUri, "GET", new Version (1, 0))) {
 				Assert.That (m.Handle, Is.Not.EqualTo (IntPtr.Zero), "Handle");
 				Assert.False (m.IsHeaderComplete, "IsHeaderComplete");
 				Assert.True (m.IsRequest, "IsRequest");
 				Assert.Throws<InvalidOperationException> (delegate { var x = m.ResponseStatusCode; }, "ResponseStatusCode");
 				Assert.Throws<InvalidOperationException> (delegate { var x = m.ResponseStatusLine; }, "ResponseStatusLine");
 				Assert.That (m.Version.ToString (), Is.EqualTo ("1.0"), "Version");
-				Assert.That (TestRuntime.CFGetRetainCount (m.Handle), Is.EqualTo ((nint)1), "RetainCount");
+				Assert.That (TestRuntime.CFGetRetainCount (m.Handle), Is.EqualTo ((nint) 1), "RetainCount");
```
Comment (rolfbjarne) [inline tests/monotouch-test/CoreServices/HttpMessageTest.cs]:
```suggestion
				Assert.That (TestRuntime.CFGetRetainCount (m.Handle), Is.EqualTo ((nint)1), "RetainCount");
```

File: tests/monotouch-test/CoreServices/HttpMessageTest.cs [hunk 3]
```diff
@@ -62,14 +67,47 @@ public void CreateEmptyFalse ()
...
+						stream.EnableEvents (CFRunLoop.Main, CFRunLoop.ModeDefault);
+						stream.Open ();
+						response = await taskCompletionSource.Task;
+					}
+				}
+			}, () => done);
+			using (var auth = CFHTTPAuthentication.CreateFromResponse (response)) {
+				Assert.NotNull (auth, "Null Auth");
+				Assert.IsTrue (auth.IsValid, "Auth is valid");
+				Assert.That (TestRuntime.CFGetRetainCount (auth.Handle), Is.EqualTo ((nint)1), "RetainCount");
```
Comment (rolfbjarne) [inline tests/monotouch-test/CoreServices/HttpMessageTest.cs]:
```suggestion
				Assert.That (TestRuntime.CFGetRetainCount (auth.Handle), Is.EqualTo ((nint) 1), "RetainCount");
```

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

**Pull request merged**